inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/exporter/PrometheusExporterTest.java

```diff
@@ -0,0 +1,88 @@
+package rocks.inspectit.oce.core.exporter;
+
+import ch.qos.logback.classic.Level;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.annotation.DirtiesContext;
+import rocks.inspectit.oce.core.SpringTestBase;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class PrometheusExporterTest extends SpringTestBase {
+
+    private static final int HTTP_TIMEOUT = 1000;
+
+    static CloseableHttpClient testClient;
```

Comment:
Would we have any alternative here in comparison to the http client? Could we auto-wire the `PrometheusExporter` class and do some enabled/disabled check.. This is fine, but I ask because of other exporters that are actually having push approach so something like this can not be done..

Reply:
You could also auto wire it but then you maybe won't notice that the endpoint is not working.
For other stuff where this is not applicable I think we can use Mockito @Spy on beans or do some even simpler checks.